src/models/codap.ts

```diff
@@ -212,38 +220,39 @@ export class Codap {
             });
         });
     }
-    
-    sendDualData(data1:number[][], data1Type:string, data2:number[][], data2Type:string) {
+
+    sendDualData(dataSpecs:IDataSpec[]) {
         // if a run number has not yet been initialized, do so now.
         if (this.state.runNumber == null) {
             this.state.runNumber = 0;
         }
-        
+
         ++this.state.runNumber;
 
-        var sampleCount = Math.max(data1.length, data2.length);
-        
+        var sampleCount = Math.max(dataSpecs[0].data.length, dataSpecs[1].data.length);
+
         var items:IDataItem[] = [];
         var collection:ICollection = this.dataSetTemplate.collections[1];
-        collection.attrs[1] = {name: data1Type, type: 'numeric', precision: 4};
-        collection.attrs[2] = {name: data2Type, type: 'numeric', precision: 4};
-        
-        for(var i=0; i < sampleCount; i++) {
-            var entry1 = data1 && data1[i];
-            var entry2 = data2 && data2[i];
-            if (!entry1 && !entry2) break;
 
-            var time = entry1 ? entry1[0] : entry2[0];
-            var value1 = entry1 && entry1[1] as number;
-            var value2 = entry2 && entry2[1] as number;
-            var item:IDataItem = {Run: this.state.runNumber, Time: time};
-            if (value1 != null)
-                item[data1Type] = value1;
-            if (value2 != null)
-                item[data2Type] = value2;
+        let j, spec;
+        for(j=0; j < 2; j++) {
+            spec = dataSpecs[j];
+            collection.attrs[j+1] = {name: spec.name, type: 'numeric', unit: spec.unit, precision: 4};
+        }
+
+        for(var i=0; i < sampleCount; i++) {
+            let item:IDataItem = {Run: this.state.runNumber };
+            for(j=0; j < 2; j++) {
+                let spec = dataSpecs[j],
+                    sample = spec.data[i];
+                if (item.Time == null)
+                    item.Time = sample[0];
+                if (sample[1] != null)
```

Comment:
I added this check to match what had been added to master from another PR. However, I wasn't sure if it was needed anymore.